src/cmd/troubleshoot/image.go

```diff
@@ -192,83 +187,73 @@ func checkCustomModuleImagePullable(httpClient *http.Client, componentName strin
 
-func addProxy(troubleshootCtx *troubleshootContext) error {
-	proxyUrl := ""
-	if troubleshootCtx.dynakube.Spec.Proxy != nil {
-		if troubleshootCtx.dynakube.Spec.Proxy.Value != "" {
-			proxyUrl = troubleshootCtx.dynakube.Spec.Proxy.Value
-		} else if troubleshootCtx.dynakube.Spec.Proxy.ValueFrom != "" {
-			var err error
-			proxyUrl, err = kubeobjects.ExtractToken(&troubleshootCtx.proxySecret, dtclient.CustomProxySecretKey)
-			if err != nil {
-				return errorWithMessagef(err, "failed to extract proxy secret field")
-			}
-		}
-	}
-	if proxyUrl != "" {
-		p, err := url.Parse(proxyUrl)
-		if err != nil {
-			return errorWithMessagef(err, "could not parse proxy URL!")
-		}
-		t := troubleshootCtx.httpClient.Transport.(*http.Transport)
-		t.Proxy = http.ProxyURL(p)
-		logInfof("using  '%s' proxy to connect to the registry", p.Host)
+	if err != nil {
+		return errors.Wrapf(err, "registry '%s' unreachable", registry)
+	} else if statusCode != http.StatusOK {
+		// Don't fail immediately since connection works.
+		// Maybe registry is not correctly implemented.
+		logWarningf("registry '%s' is reachable but returned an unexpected error code (%d)", registry, statusCode)
 	}
 
 	return nil
 }
 
-func connectToDockerRegistry(httpClient *http.Client, httpMethod string, httpUrl string, authMethod string, authToken string) (int, error) {
-	body := strings.NewReader("")
+func connectToDockerRegistry(httpClient *http.Client, httpUrl string, authToken string) (int, error) {
+	req, err := http.NewRequest("HEAD", httpUrl, nil)
 
-	req, err := http.NewRequest(httpMethod, httpUrl, body)
 	if err != nil {
 		return 0, err
 	}
-	if authMethod != "" && authToken != "" {
-		req.Header.Set("Authorization", authMethod+" "+authToken)
+
+	if authToken != "" {
+		req.Header.Set("Authorization", "Basic"+" "+authToken)
 	}
 
 	resp, err := httpClient.Do(req)
 	if err != nil {
 		return 0, err
 	}
-	defer resp.Body.Close()
 
-	_, err = io.ReadAll(resp.Body)
-	if err != nil {
-		return 0, err
-	}
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+
 	return resp.StatusCode, nil
 }
 
-func getPullSecretToken(troubleshootCtx *troubleshootContext) (string, error) {
-	secretBytes, ok := troubleshootCtx.pullSecret.Data[dtpullsecret.DockerConfigJson]
-	if !ok {
-		return "", fmt.Errorf("token .dockerconfigjson does not exist in secret '%s'", troubleshootCtx.pullSecretName)
+func getPullSecret(troubleshootCtx *troubleshootContext) (string, error) {
+	secretBytes, hasPullSecret := troubleshootCtx.pullSecret.Data[dtpullsecret.DockerConfigJson]
+	if !hasPullSecret {
+		return "", fmt.Errorf("token .dockerconfigjson does not exist in secret '%s:%s'",
+			troubleshootCtx.pullSecret.Namespace, troubleshootCtx.pullSecret.Name)
 	}
 
-	secretStr := string(secretBytes)
-	return secretStr, nil
+	return string(secretBytes), nil
+}
+
+func manifestUrl(registry string, componentImageInfo imageInfo) string {
```

Comment:
can these helper functions be combined with https://github.com/Dynatrace/dynatrace-operator/blob/master/src/api/v1beta1/image_paths.go?

Reply:
these helper functions depend on imageInfo which is created by parsing the image uri. Could be combined, but only if the parsing logic is also moved and incorporated into `image_paths.go`. Seems like a lot of work for functions that just put a string together

Reply:
refactored